### PR TITLE
 Fix gem declarations for Monolithic builds

### PR DESCRIPTION
**Pull request by wc-robotec**
### **Gems build properly on engine version 2409**

Replaced
```bash
#if defined(O3DE_GEM_NAME)
AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME, _Editor), FOO::FOO)
#else
AZ_DECLARE_MODULE_CLASS(Gem_FOO FOO::FOO)
#endif
```
Into
```bash
AZ_DECLARE_MODULE_CLASS(Gem_FOO FOO::FOO)
```

Files affected:
- ExposeConsoleToRosModule.cpp
- ExposeConsoleToRosEditorModule.cpp
- SmoothingEditorModule.cpp

### CODE REVIEW

File: Gems/ExposeConsoleToRos/Code/Source/Clients/ExposeConsoleToRosModule.cpp [hunk 1]
```diff
@@ -13,8 +13,4 @@ namespace ExposeConsoleToRos
     };
 } // namespace ExposeConsoleToRos
 
-#if defined(O3DE_GEM_NAME)
-AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME), ExposeConsoleToRos::ExposeConsoleToRosModule)
-#else
-AZ_DECLARE_MODULE_CLASS(Gem_ExposeConsoleToRos, ExposeConsoleToRos::ExposeConsoleToRosModule)
-#endif
+AZ_DECLARE_MODULE_CLASS(Gem_ExposeConsoleToRos, ExposeConsoleToRos::ExposeConsoleToRosModule)
```
Comment (pijaro) [inline Gems/ExposeConsoleToRos/Code/Source/Clients/ExposeConsoleToRosModule.cpp:16]: Keep newline please

File: Gems/ExposeConsoleToRos/Code/Source/Tools/ExposeConsoleToRosEditorModule.cpp [hunk 2]
```diff
@@ -37,8 +37,4 @@ namespace ExposeConsoleToRos
     };
 } // namespace ExposeConsoleToRos
 
-#if defined(O3DE_GEM_NAME)
-AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME, _Editor), ExposeConsoleToRos::ExposeConsoleToRosEditorModule)
-#else
-AZ_DECLARE_MODULE_CLASS(Gem_ExposeConsoleToRos_Editor, ExposeConsoleToRos::ExposeConsoleToRosEditorModule)
-#endif
+AZ_DECLARE_MODULE_CLASS(Gem_ExposeConsoleToRos_Editor, ExposeConsoleToRos::ExposeConsoleToRosEditorModule)
```
Comment (pijaro) [inline Gems/ExposeConsoleToRos/Code/Source/Tools/ExposeConsoleToRosEditorModule.cpp:40]: Keep newline please

File: Gems/Smoothing/Code/Source/Tools/SmoothingEditorModule.cpp [hunk 3]
```diff
@@ -38,8 +38,4 @@ namespace Smoothing
     };
 } // namespace Smoothing
 
-#if defined(O3DE_GEM_NAME)
-AZ_DECLARE_MODULE_CLASS(AZ_JOIN(Gem_, O3DE_GEM_NAME, _Editor), Smoothing::SmoothingEditorModule)
-#else
-AZ_DECLARE_MODULE_CLASS(Gem_Smoothing_Editor, Smoothing::SmoothingEditorModule)
-#endif
+AZ_DECLARE_MODULE_CLASS(Gem_Smoothing_Editor, Smoothing::SmoothingEditorModule)
```
Comment (pijaro) [inline Gems/Smoothing/Code/Source/Tools/SmoothingEditorModule.cpp:41]: Keep newline please

File: .gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
 .vscode/**
+.idea/**
```
Comment (pijaro) [inline .gitignore:2]: Not in this PR

**Review by pijaro — CHANGES_REQUESTED** [review]

**Comment by wc-robotec** [comment]
Added new line at the end of affected files. 
I've removed gitignore changes, I'll create seperate PR for this

**Review by pijaro — APPROVED** [review]

**Pull request merged**